### PR TITLE
Show all Expensify logins in Contact Methods page & view each in new Contact Method Details page

### PR DESCRIPTION
**Pull request by Beamanator**
cc @cristipaval since you're helping with all of these related account settings / contact methods pages

- ~Holding on https://github.com/Expensify/App/pull/15039, which this PR was forked from~
- ~Holding on https://github.com/Expensify/Web-Expensify/pull/36541, where all the new commands will be created~

### Details
Nothing too fishy, but I left a few comments about places we have temporary logic that we'll remove soon - the next phase in this rollout is to create the "Add new contact method" page. At this point, we'll remove the old `LoginField` component & the old `AddSecondaryLoginPage` page and route

### Fixed Issues
$ https://github.com/Expensify/App/issues/15203
$ https://github.com/Expensify/App/issues/11575

### Tests

Notes:
1. If you hit a rate limiting / throttling error when adding secondary logins, use this to delete the throttle:
    - `DELETE FROM nameValuePairs WHERE accountID = <accountID> and name glob 'private_throttleAttempts_*';`
2. This web-e PR is needed for testing: https://github.com/Expensify/Web-Expensify/pull/36541

#### Default contact method
1. Navigate to Settings -> Profile -> Contact method
2. Select your default / primary contact method
3. Verify you only see appropriate text on that page

Screenshot 2023-03-01 at 5 05 37 PM

#### Still able to add new contact methods
1. Log in to an account that only has an email login OR only has a phone number login
2. Navigate to Settings -> Profile -> Contact method
3. Verify you can see your default contact method AND:
    1. If your default contact method is an email address, you can see a place to add a phone number login
    2. Vise versa
    - Screenshot 2023-03-01 at 5 06 25 PM
4. Verify adding a new contact method still works, with your old password (we're removing the password flow soon)
5. Verify you now see this unverified login in the "Contact methods" page, with a green dot next to it
    - Screenshot 2023-03-01 at 5 07 45 PM
6. Verify you also see a green dot in these places:
    -  | Main Settings button | On Settings page | On Profile page |
        |---|---|---|
        | Screenshot 2023-02-27 at 6 26 24 PM | Screenshot 2023-02-27 at 6 27 03 PM | Screenshot 2023-02-27 at 6 27 28 PM | 

---

#### Resend magic code to unvalidated contact method & validate it
1. Navigate to Settings -> Profile -> Contact method
2. View the unverified contact method you just added in the previous step
3. Verify you see a page like this:
    - Screenshot 2023-03-01 at 5 08 16 PM
4. Click "Resend magic code"
5. Verify you can now see a green checkmark next to the "Resend magic code" link
    - Screenshot 2023-03-01 at 5 08 37 PM
6. Verify you can successfully validate the login by entering `000000` magic code (@cristipaval will make the link work in a future PR)
    -  Note: You should get automatically navigated back to the contact methods page
7. Navigate back to the login you just validated, verify you can only see a "Remove" button on that page now
    - Note: In the near future we'll also add a "Set contact method as default" button, which will make the page look less dreary
    - (this is the view for a verified contact method that's not your default contact method)

#### Resend magic code, and observe errors
1. Navigate to Settings -> Profile -> Contact method
2. View an unverified contact method
3. Click "Resend magic code" many times. Verify you eventually see this error:
    - Screenshot 2023-03-01 at 5 09 18 PM
4. Navigate back to "Contact methods" and verify a red dot now shows up on that contact method row
    - You should also see a red dot in the main Settings indicator in the LHN, on the Settings page, and on the Profile page
5. Navigate to that contact method details page, click the red "X", verify the error goes away

#### Validate secondary login, and observe errors
1. Navigate to Settings -> Profile -> Contact method
2. View an unverified contact method
3. Enter an invalid magic code like `12345`. Click "Verify".
4. Verify you are redirected back to the Contact methods page and you quickly (once the request fails) see a red dot on the secondary login you tried to verify
6. Open that contact method detail page, verify you see an error like this:
    - Screenshot 2023-03-01 at 5 09 49 PM
    - You should also see a red dot in the main Settings indicator in the LHN, on the Settings page, and on the Profile page
7. Verify clicking the "X" clears the error

---

#### Remove contact method
1. Navigate to Settings -> Profile -> Contact method
7. View a non-default contact method (doesn't matter if it's verified yet or not)
10. Click "Remove", verify a popup shows up
    - Screenshot 2023-02-23 at 5 10 19 PM
11. Click "No", verify you go back to the contact method detail screen as before.
12. Try again, but click "Yes"
13. Verify you're taken back to the Contact methods page, and that contact method was removed.

- [X] Verify that no errors appear in the JS console

#### Send magic code
1. Navigate to Settings -> Profile -> Contact method
2. View an unverified contact method
3. Click "Resend magic code"
4. Verify the UI looks like this while offline:

Screenshot 2023-03-01 at 5 11 19 PM

#### Validate a secondary login
1. Navigate to Settings -> Profile -> Contact method
2. View an unverified contact method
3. Enter any magic code, then click "Verify"
4. You should get routed back to the contact methods page. Navigate in to that contact method again
5. Verify the UI looks like this while offline:

Screenshot 2023-03-01 at 5 11 49 PM

#### Delete a secondary login
1. Navigate to Settings -> Profile -> Contact method
2. View an unverified contact method
3. Click "Remove" while offline
4. Verify you're back on the "Contact methods" page, and the login you're trying to delete looks like this (not necessarily with the red dot, if your contact method was validated previoiusly):

Screenshot 2023-03-01 at 5 12 05 PM

### QA Steps
Same as above, except:
1. When validating a contact method with a magic code, don't use `000000` - use the code that is sent to your email

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/3885503/221638255-cb4a221b-2b79-4782-9b59-c330f9ca4fe5.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/3885503/222133669-a621b291-cb81-46cd-95eb-f961298ab0ad.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/3885503/222138546-304da173-e438-48b1-b8cc-c5be7261a266.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/3885503/221642089-6a9f0311-9135-4fde-b091-4868b8f9a255.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/3885503/222137317-dc5b927d-c604-4964-a6fc-7d97259fc4dc.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/3885503/222132844-7d75dbb6-a866-445c-8eaf-e3c8373d7f48.mov

</details>

### CODE REVIEW

**Comment by Beamanator** [comment]
@shawnborton I'm requesting you review this too if you don't mind, at least the screenshots I posted 🙏

File: src/styles/styles.js [hunk 1]
```diff
@@ -1016,24 +1016,7 @@ const styles = {
...
-        bottom: 4,
-        height: 16,
-        width: 16,
-        zIndex: 10,
-    },
-
-    statusIndicatorOnline: {
-        backgroundColor: themeColors.success,
-    },
+    }),
```
Comment (Beamanator) [inline src/styles/styles.js:1019]: Deleting this cuz it turns out this is not being used by AvatarWithIndicator (or anywhere)

**Comment by shawnborton** [comment]
For this screen here:
image


It does feel weird that the remove row is kinda floating on its own yet so close to the form above right?

**Comment by Beamanator** [comment]
> Can you check the left padding for all of your screenshots? This should have been fixed by @cristipaval recently

Gooood call, that was just updated veryyyyy recently 😅 

> For this screenshot (re: default contact method) if you can't remove it, why even bother showing the disabled row? I would just put the text at the very top and skip the disabled row I think.

I am down to completely hide it at that point 👍  I just think in previous mockups we had it showing and disabled, but I like your idea better! (since I don't think we normally like showing things you can't interact with)

> (RE: the remove button on a page with an unvalidated contact method) It does feel weird that the remove row is kinda floating on its own yet so close to the form above right?

I would say it doesn't look too weird to me, but we can definitely consider changing how it looks it if you prefer!

**Comment by Beamanator** [comment]
@shawnborton Updated screenshots, let me know how they're lookin! 👍

**Comment by shawnborton** [comment]
Minor - looks like a weird word-break here:
image

**Comment by shawnborton** [comment]
For this:
image


I wonder if we should make the green dot on settings match the size of the green dot elsewhere? That means it should be like 12px x 12px.

**Comment by shawnborton** [comment]
A couple more follow ups...


Can you show me what this screen looks like when the account is verified?
image


Maybe we can add like 12-20px more top margin to the remove button so there is more space between it and the verify form. 

Also, for the remove row, I wonder if we should make that icon red? cc @trjExpensify  - I think we are doing that in a lot of the N7 docs. Basically any option row that does a delete action should use red for the icon color.

**Comment by trjExpensify** [comment]
>Also, for the remove row, I wonder if we should make that icon red? cc @trjExpensify - I think we are doing that in a lot of the N7 docs. Basically any option row that does a delete action should use red for the icon color.

Yep, agreed. We're using red for the delete action option rows in N7, so we should do the same here for `Remove`.

**Comment by Beamanator** [comment]
@shawnborton here's what I think it could look like if we want the green circle to be 12px by 12px - do you like?

Screenshot 2023-03-02 at 1 40 34 PM
Screenshot 2023-03-02 at 1 40 42 PM

**Comment by Beamanator** [comment]
@shawnborton 

> Can you show me what this screen looks like when the account is verified?

Here's what it currently looks like - and I noted above that I'm planning to add a "Set contact method as default" button (don't remember the exact copy) in a follow-up PR, since that change requires another kinda complicated command 

https://user-images.githubusercontent.com/3885503/222420388-91169989-99cb-4bda-b016-848039ed3668.mov

> Maybe we can add like 12-20px more top margin to the remove button so there is more space between it and the verify form.

Sure! Let me know which of these looks better:

| 12px added | 20px added |
|---|---|
| Screenshot 2023-03-02 at 1 52 14 PM | Screenshot 2023-03-02 at 1 52 44 PM |

**Comment by shawnborton** [comment]
Cool, thanks for confirming. Tag me when you add that "Set as default" button! 

For the spacing, I think I like 20px more.

**Comment by shawnborton** [comment]
Well that's because we were just reusing the old online status position :)

**Comment by Beamanator** [comment]
Hahaha fair 😅  How do you like this, @shawnborton ?

Screenshot 2023-03-02 at 8 19 43 PM

**Comment by Beamanator** [comment]
FYI @cristipaval @mollfpr I'll be OOO tomorrow (Friday) but will be back at least half day Sunday to make up for it - so if y'all cannn do at least a partial review tomorrow (Friday) that would be very helpful, as this is an important PR & is holding up at least 2 others (1 from @cristipaval and one final future PR)

**Comment by shawnborton** [comment]
Looks good to me, thank you!

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

File: src/styles/styles.js [hunk 2]
```diff
@@ -1006,35 +1006,18 @@ const styles = {
...
-        borderColor: themeColors.componentBG,
-        backgroundColor: themeColors.danger,
+        backgroundColor,
         borderRadius: 8,
         borderWidth: 2,
         position: 'absolute',
-        right: 4,
-        bottom: 4,
+        right: -2,
+        top:-1,
```
Comment (mollfpr) [inline src/styles/styles.js:1016]:
```suggestion
        top: -1,
```

**Review by mollfpr — COMMENTED** [review]
Well done on this PR @Beamanator 👍 

I Left a small comment, and everything else looks good!

**Comment by mollfpr** [comment]
@Beamanator I found a weird behavior on mWeb/Chrome when dismissing the keyboard; we can see the chat list for a moment.

https://user-images.githubusercontent.com/25520267/222907823-e6258735-ff04-4650-af5e-e884b183a798.mp4

I'm testing on Samsung ZFlip. It is not reproducible on mWeb/Safari, mWeb/Chrome on staging, and mWeb/Chrome Android simulator.

https://user-images.githubusercontent.com/25520267/222908108-604cf3dc-8c9a-4001-87d2-7e0dd077a2b4.mov

File: src/styles/styles.js (hunk 2 above)
Comment (Beamanator) [reply src/styles/styles.js:1016]: Nice, I wonder why eslint didn't catch that!

**Comment by Beamanator** [comment]
@mollfpr thanks for the comments, I addressed them! 👍 

About that weird issue when adding a secondary login, I won't worry about that for now since I'm not modifying that page (`AddSecondaryLoginPage`) in this PR, but maybe you can try to reproduce on other pages and report a separate bug if it's reproducible!

**Comment by mollfpr** [comment]
@Beamanator Yeah, I feel this PR has nothing to do about the issue. I'll take a look more at issue and report it. Besides that, everything looks good to me 👍

**Review by mollfpr — APPROVED** [review]
Well done @Beamanator 🚀

**Comment by Beamanator** [comment]
Groovy, thanks @mollfpr 👍 

@cristipaval @shawnborton all y'all! 👍 😈

**Comment by shawnborton** [comment]
Looks good to me!

**Review by cristipaval — APPROVED** [review]

**Comment by Beamanator** [comment]
FYI this PR got reverted in this PR (https://github.com/Expensify/App/pull/15761) due to this bug that it caused: https://github.com/Expensify/App/issues/15725

File: src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                    {!loginData.validatedDate && (
+                        <View style={[styles.mh5, styles.mb7]}>
+                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.mb1, styles.mt3]}>
+                                <Icon src={Expensicons.DotIndicator} fill={colors.green} />
+                                <View style={[styles.flex1, styles.ml4]}>
+                                    <Text>
+                                        {this.props.translate('contacts.enterMagicCode', {contactMethod})}
+                                    </Text>
+                                </View>
+                            </View>
```
[the remark this answers is not in the document]
Comment (Beamanator) [reply src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js:177]: @shawnborton approved the designs of this PR, are you sure we wanted the other designs there? It looks like we didn't wait for Shawn to respond in Slack - https://expensify.slack.com/archives/C049HHMV9SM/p1683310488445319

And he wasn't ping'ed in that issue 🤷

[the remark this answers is not in the document]
Comment (Beamanator) [reply src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js:177]: Ooh perfecto, thanks @0xmiroslav !

**Pull request merged**